### PR TITLE
C4 teleports into pocket when activated outside the pocket #64308

### PR DESCRIPTION
**Pull request by aahue1889**
#### Summary
Bugfixes "#64308"

#### Purpose of change

C-4 teleports into Pocket after activating off of floor, 

Steps to Repro:

> ### Steps to reproduce
> 1. Spawn C-4
> 2. Drop it
> 3. `a`ctivate it when it's on the ground
> 4. The game allow you to activate it
> 5. After activation the item teleport into your pockets
> 
> 
> ### Screenshots
> GIF 16-03-2023 17-57-56
> 

#### Describe the solution

Changed Flags for Json Objects C-4 

image

The Allows Remote Use/ Radio Modable/ Radio Invoke Proc/ Bomb tags have been added to C4
- This makes it behave similar to a "Gas Drum' Bomb; AND allows for remote control detonation usage

#### Describe alternatives you've considered

I was considering making changes on the C++ side, but when I started adding flags, I noticed that the behavior of C4 was still respected, so I didn't see a need of changing anything in the underlying C++ code located here:

image

#### Testing

I was able to confirm with C4 that I was able to place them and activate without them warping to your pocket
- To Test Allow_Remote_Use:  I performed the following test 
https://user-images.githubusercontent.com/8581536/225964275-9c285bdb-69b6-46eb-bb11-5fe3b9dfcafa.mp4
- To Test Radio Invoke and Radio modable 
  - I spawned in a RadioMod and attached it to a RC Controller  and another RadioMod to Attach the the C4 (Signal Red)
  - Push the red controller button, and you will be prompted to set the c4 timer and it will explode 
- To test BOMB flag 
  - All I did was just make sure the item was throwable; these items are heavy so they can only be thrown 1-2 squares; but if you want to toss the items you can now.

### CODE REVIEW

**Comment by GuardianDll** [comment]
Want to point once again, that adding ALLOWS_REMOTE_USE also allows player to activate the item inside the backpack (for demolition charge you have no option to confirm, the action is performed silently), which may cause more harm

**Comment by Terrorforge** [comment]
> Want to point once again, that adding ALLOWS_REMOTE_USE also allows player to activate the item inside the backpack (for demolition charge you have no option to confirm, the action is performed silently), which may cause more harm

Are you talking about a backpack on the ground? Because it is already possible to active C-4 that's just sitting in your inventory.

**Comment by aahue1889** [comment]
I'll remove the code changes to DemoCharge

File: data/json/items/tool/explosives.json [hunk 1]
```diff
@@ -1484,7 +1484,7 @@
       "menu_text": "Light fuse",
       "type": "transform"
     },
-    "flags": [ "RADIO_MODABLE", "RADIO_INVOKE_PROC", "BOMB" ]
+    "flags": [ "RADIO_MODABLE", "RADIO_INVOKE_PROC", "BOMB", "ALLOWS_REMOTE_USE" ]
```
Comment (aahue1889) [inline data/json/items/tool/explosives.json:1487]:
```suggestion
    "flags": [ "RADIO_MODABLE", "RADIO_INVOKE_PROC", "BOMB"]
```

**Review by aahue1889 — COMMENTED** [review]
Reverting the code change for homemade_demo charge

**Pull request merged**